packages/jellyfish-api-core/__tests__/category/loan/depositToVault.test.ts

```diff
@@ -0,0 +1,329 @@
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import { GenesisKeys } from '@defichain/testcontainers'
+import BigNumber from 'bignumber.js'
+import { TestingGroup } from '@defichain/jellyfish-testing'
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+
+describe('Loan', () => {
+  const tGroup = TestingGroup.create(2, i => new LoanMasterNodeRegTestContainer(GenesisKeys[i]))
+  let vaultId: string
+  let vaultId1: string
+  let liqVaultId: string
+  let collateralAddress: string
+  let vaultAddress: string
+
+  beforeAll(async () => {
+    await tGroup.start()
+    await tGroup.get(0).container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterAll(async () => {
+    await tGroup.stop()
+  })
+
+  async function setup (): Promise<void> {
+    // token setup
+    collateralAddress = await tGroup.get(0).container.getNewAddress()
+    await tGroup.get(0).token.dfi({ address: collateralAddress, amount: 30000 })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).token.create({ symbol: 'BTC', collateralAddress })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).token.mint({ symbol: 'BTC', amount: 20000 })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).token.create({ symbol: 'CAT', collateralAddress })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).token.mint({ symbol: 'CAT', amount: 10000 })
+    await tGroup.get(0).generate(1)
+
+    // oracle setup
+    const addr = await tGroup.get(0).generateAddress()
+    const priceFeeds = [
+      { token: 'DFI', currency: 'USD' },
+      { token: 'BTC', currency: 'USD' },
+      { token: 'TSLA', currency: 'USD' },
+      { token: 'CAT', currency: 'USD' }
+    ]
+    const oracleId = await tGroup.get(0).rpc.oracle.appointOracle(addr, priceFeeds, { weightage: 1 })
+    await tGroup.get(0).generate(1)
+    const timestamp = Math.floor(new Date().getTime() / 1000)
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1@DFI', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '10000@BTC', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '2@TSLA', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '10000@CAT', currency: 'USD' }] })
+    await tGroup.get(0).generate(1)
+
+    // collateral token
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      priceFeedId: 'DFI/USD'
+      // activateAfterBlock: 130  // <- hit socket hang up
```

Comment:
@canonbrother thanks for spotting this. I know why this is happening. I need to write some cpp side testing to narrow it down. will address this separately.